public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -237,110 +272,141 @@ const Policy: FC<PolicyComponentProps> = ({
+              // This child is autogenerated if it's the autogenerated root or if it's a child of an autogenerated policy.
+              const isThisChildAutoGenerated = isAutoGeneratedRootAndSimplifiedEnabled(child) || isAutoGenerated;
+              /* pass the "readOnly" prop from the parent, because for any child policy , if its parent it's not editable, 
+              then the child policy should not be editable either */
+              const isThisChildReadOnly = readOnly || provisioned || isAutoGenerated;
+
+              return (
+                <Policy
+                  key={uniqueId()}
+                  routeTree={routeTree}
+                  currentRoute={child}
+                  receivers={receivers}
+                  contactPointsState={contactPointsState}
+                  readOnly={isThisChildReadOnly}
+                  inheritedProperties={childInheritedProperties}
+                  onAddPolicy={onAddPolicy}
+                  openDetailModal={openDetailModal}
+                  onDeletePolicy={onDeletePolicy}
+                  onShowAlertInstances={onShowAlertInstances}
+                  alertManagerSourceName={alertManagerSourceName}
+                  alertGroups={alertGroups}
+                  routesMatchingFilters={routesMatchingFilters}
+                  matchingInstancesPreview={matchingInstancesPreview}
+                  isAutoGenerated={isThisChildAutoGenerated}
+                  isAutogeneratedChunkOpen={isAutogeneratedChunkOpen}
+                  toggleAutogeneratedChunk={toggleAutogeneratedChunk}
+                />
+              );
+            })}
+          </>
+        </div>
+        {showExportDrawer && <GrafanaPoliciesExporter onClose={toggleShowExportDrawer} />}
+      </Stack>
+    </ConditionalWrap>
+  ) : null;
+};
 
-          return (
-            <Policy
-              key={uniqueId()}
-              routeTree={routeTree}
-              currentRoute={child}
-              receivers={receivers}
-              contactPointsState={contactPointsState}
-              readOnly={readOnly || provisioned}
-              inheritedProperties={childInheritedProperties}
-              onAddPolicy={onAddPolicy}
-              onEditPolicy={onEditPolicy}
-              onDeletePolicy={onDeletePolicy}
-              onShowAlertInstances={onShowAlertInstances}
-              alertManagerSourceName={alertManagerSourceName}
-              alertGroups={alertGroups}
-              routesMatchingFilters={routesMatchingFilters}
-              matchingInstancesPreview={matchingInstancesPreview}
-            />
-          );
-        })}
-      </div>
-      {showExportDrawer && <GrafanaPoliciesExporter onClose={toggleShowExportDrawer} />}
-    </Stack>
+export const AUTOGENERATED_ROOT_LABEL_NAME = '__grafana_autogenerated__';
+
+export function isAutoGeneratedRootAndSimplifiedEnabled(route: RouteWithID) {
```

Comment:
Maybe `isAutoGeneratedRoot` is enough for the name? If the feature toggle is disabled there won't be autogenerated root

Reply:
At the beginning I named this way, but then I realised that this function not only checks if it's autogenerated...it also checks if toggle is enabled....once we remove the toggle this function will be renamed as you suggest